src/traces/indicator/plot.js

```diff
@@ -0,0 +1,788 @@
+    var fgArcPath = fgArc.select('path');
+    if(hasTransition) {
+        fgArcPath
+            .transition()
+            .duration(transitionOpts.duration)
+            .ease(transitionOpts.easing)
+            .each('end', function() { trace._lastValue = cd[0].y; onComplete && onComplete(); })
+            .each('interrupt', function() { onComplete && onComplete(); })
+            .attrTween('d', arcTween(valueArcPath, valueToAngle(cd[0].lastY), valueToAngle(cd[0].y)));
+    } else {
+        fgArcPath
+            .attr('d', valueArcPath.endAngle(valueToAngle(cd[0].y)));
+    }
+    fgArcPath.call(styleShape);
+    fgArc.exit().remove();
+    var gaugeBorder = angularGauge.selectAll('g.gaugeOutline').data([gaugeOutline]);
+    gaugeBorder.enter().append('g').classed('gaugeOutline', true).append('path');
+    gaugeBorder.select('path').call(drawArc).call(styleShape);
+    gaugeBorder.exit().remove();
+}
+
+function drawNumbers(gd, plotGroup, cd, opts) {
+    var trace = cd[0].trace;
+    var numbersX = opts.numbersX;
+    var numbersY = opts.numbersY;
+    var numbersAnchor = anchor[trace.align || 'center'];
+
+    var hasTransition = opts.hasTransition;
+    var transitionOpts = opts.transitionOpts;
+    var onComplete = opts.onComplete;
+
+    var bignumberFontSize, deltaFontSize;
+    if(trace._hasNumber) bignumberFontSize = trace.number.font.size;
+    if(trace._hasDelta) deltaFontSize = trace.delta.font.size;
+
+    // Position delta relative to bignumber
+    var deltaDy = 0;
+    var deltaX = 0;
+    var bignumberY = 0;
+
+    if(trace._hasDelta && trace._hasNumber) {
+        if(trace.delta.position === 'bottom') {
+            deltaDy = deltaFontSize * 1.5;
+        }
+        if(trace.delta.position === 'top') {
+            deltaDy = -bignumberFontSize + MID_SHIFT * deltaFontSize;
+        }
+        if(trace.delta.position === 'right') {
+            deltaX = undefined;
+        }
+        if(trace.delta.position === 'left') {
+            deltaX = undefined;
+            bignumberY = MID_SHIFT * bignumberFontSize / 2;
+        }
+    }
+    deltaDy -= MID_SHIFT * deltaFontSize;
+
+    var numbers = plotGroup.selectAll('text.numbers').data(cd);
+    numbers.enter().append('text').classed('numbers', true);
+
+    var data = [];
+    if(trace._hasNumber) data.push('number');
+    if(trace._hasDelta) {
+        data.push('delta');
+        if(trace.delta.position === 'left') data.reverse();
+    }
+    var sel = numbers.selectAll('tspan').data(data);
+    sel.enter().append('tspan');
+    sel
+        .attr('text-anchor', function() {return numbersAnchor;})
+        .attr('class', function(d) { return d;})
+        .attr('dx', function(d, i) {
+            // Add padding to the second tspan when it's a one-liner
+            if(i === 1) {
+                var pos = trace.delta.position;
+                if(pos === 'left' || pos === 'right') return 10;
+            }
+            return null;
+        });
+    sel.exit().remove();
```

Comment:
You'll need a `sel.order()` call to make sure that

```js
Plotly.newPlot(gd, [{mode: 'number+delta'}])
```

has the same `<tspan>` order as after

```js
Plotly.restyle(gd, 'mode', 'delta')
Plotly.restyle(gd, 'mode', 'number+delta')
```

Reply:
I test this in 95de45724b62a7a1c3d22434af838bd8c690cbfe and it seems to be OK without a `sel.order()`. To be investigated

Reply:
The test look fine. Thanks for adding those! Something else must up with a upstream `.data()` call.

Oh well. No need to waste time taking a deeper dive at this problem. :ok_hand: